---
title: "Functional Metagenomics analysis using Phyloseq"
author: "Lucía Martín Fernández"
output:
  html_document:
    code_folding: hide
    fig_caption: yes
    number_sections: yes
    theme: paper
    toc: true
    toc_float: yes
  html_notebook:
    code_folding: hide
    fig_caption: yes
    number_sections: yes
    theme: paper
    toc: true
    toc_float: yes
  pdf_document:
    toc: yes
subtitle: Bioinformatics Results
---

# Install Packages
```{r setup, include=FALSE}

# install.packages("tidyverse")
# install.packages('devtools')
# install.packages("BiocManager")
# BiocManager::install("phyloseq")
# BiocManager::install("DESeq2")
# remotes::install_github("KlausVigo/phangorn")
# remotes::install_github("cpauvert/psadd")
# install_github("guokai8/microbial")
#install_github("Russel88/MicEco")
```

# Load Packages
```{r Load Packages, include=FALSE}

library("tidyverse")
library("BiocManager")
library("ggplot2")      # graphics
library("dplyr")        # filter and reformat data frames  
library("phyloseq")
library("readxl")       # necessary to import the data from Excel file
library("tibble")
library("vegan")
library("DESeq2")
library("dendextend") 
library("psadd")
library("microbial")
library ("MicEco")
library("ggpubr")
```

# Load the data

The output ko_abundance.tsv file was open with Excel to create an Excel file (ko_abundance.xlsx) with all three necessary data groups to work with Phyloseq objects:
- *TPM_df* contains the relative abundance of each KO in the sample calculated in TPM. This information is in the 'TPM' sheet of the 'ko_abundance.xlsx' file. 
- *KO_df* contains the KO ids together with its description and symbol. This information is in the 'KO' sheet of the 'ko_abundance.xlsx' file. 
- *samples_df* contains the metadata of each sample. This information is in the 'metadata' sheet of the 'ko_abundance.xlsx' file. 

```{r Loading  data, include=TRUE}

#Loading the data
setwd("~/Desktop/RyC_TFM/final_results")

#ko_abundance <- read_excel("ko_abundance.xlsx",  sheet = "KO")
TPM_df <- read_excel("ko_totalabundance.xlsx",  sheet = "TPM")

#ko_abundance <- read_excel("ko_abundance.xlsx", sheet = "TPM")
KO_df  <- read_excel("ko_totalabundance.xlsx", sheet = "KO")

samples_df  <- read_excel("ko_totalabundance.xlsx", sheet = "metadata")

# We want to add a new samples variable to merge the subjects' Group (Donor, Fecal Microbial Transplant (FMT) or Placebo) with the data collecting Weeks variables
samples_df$group_week <- paste (samples_df$Group, samples_df$week, sep = "_")
```

# Create phyloseq object

A Phyloseq object is created with the data stored in the previously loaded dataframes that are firstly transformed to matrices when needed. 

```{r Create phyloseq object, include=TRUE}

# Define the row names from the KEGG_ko column in the TPM_df
TPM_df <- TPM_df %>%
  tibble::column_to_rownames("KEGG_ko")

TPM_df[is.na(TPM_df)] <- 0

# Idem for the other dataframes
KO_df <- KO_df %>% 
  tibble::column_to_rownames("KEGG_ko")

samples_df <- samples_df %>% 
  tibble::column_to_rownames("Sample_ID")

#Transform into matrices TPM and KO tables (sample table can be left as data frame)
TPM_mat <- as.matrix(TPM_df)

KO_mat <- as.matrix(KO_df)

class(TPM_mat) <- "numeric" # Because it contains the TPM calculated for each KO. We are going to consider this values as normalized read counts for each KO

# Transform to Phyloseq objects. 
# Since PhyloSeq is usually implemented for taxonomy and here we are working with functional annotations, some equivalencies are made:
TPM = otu_table(TPM_mat, taxa_are_rows = TRUE) 
KO = tax_table(KO_mat)
samples = sample_data(samples_df)

raw_ps <- phyloseq(TPM, KO, samples)

# We want to normalize the number of normalized reads so it can be compare between samples. For this, we are going to use the median sequencing depth. # Then, since we want read counts, we are going to round this value to have integers.  
total = median(sample_sums(raw_ps))
standf = function(x, t=total) round(t * (x / sum(x))) 

ps = transform_sample_counts(raw_ps, standf) 

# NOTE: The data we are working with does not contain the 'UNMAPPED' functions proportion. 

# We want to have the raw phyloseq object which contains normalized read counts (TPM) as integer numbers 
round_fun <- function(x) round(x)
raw_ps <- transform_sample_counts(raw_ps, round_fun) 
```

### Overview of the created Phyloseq Object

Since PhyloSeq is usually implemented for taxonomy and here we are working with functional annotations, some equivalencies are made:

Taxa in this case is refering to the KOs identifiers. Each KO identifier has a Description, Symbol and again the KO (3 taxonomic ranks) which are stored in the Taxonomy table. 
The OTU Table corresponds then to the relative abundance of each KO (taxa) in TPM units for each of the 117 samples that have 15 variables stored in the Sample Data section. 

```{r Show phyloseq object, include=TRUE}
ps # Phyloseq Object schema
ntaxa(ps) # Number of KO
nsamples(ps) # Number of samples
sample_names(ps) # All sample names
rank_names(ps) # KO's characteristics
sample_variables(ps) # Sample variables
otu_table(ps)[1:5, 1:5] # Overview of the relative abundance table of each KO in TPM
tax_table(ps)[1:5, 1:2] # Overview of the the characteristics of each KO
```

## Filtering Data

In this step we are selecting the samples we want to work with ¿¿and filtering the KO to eliminate low abundant KOs??

```{r Filtering Data, include=TRUE}

incomplete <- c("R24", "R13", "R22", "R25", "R26", "R27", "R5") # List of id patients with incomplete data


# Here we are selecting data collected at week 0 and 7 from both FMT, placebo and Donor subjects except from subjects with incomplete data
ps_subset <- subset_samples(ps, (week == "Week_0" |  week == "Week_7") & !(id %in% incomplete))
ps_subset

# Here we are selecting data collected at week 0 and 7 from subjects that received the FMT from the already subset phyloseq object
ps_FMT <- subset_samples(ps_subset, Group=="FMT") 
ps_FMT

# Here we are selecting data collected at week 0 and 7 from Placebo subjects from the already subset phyloseq object
ps_placebo <- subset_samples(ps_subset, Group=="Placebo")
ps_placebo

# Here we are selecting the Donor samples (there is only information collected at week 0)
ps_donor <- subset_samples(ps, Group=="Donor")

ps_raw_subset <- subset_samples(raw_ps, (week == "Week_0" |  week == "Week_7") & !(id %in% incomplete))
ps_raw_FMT <- subset_samples(ps_raw_subset, Group=="FMT")
ps_raw_placebo <- subset_samples(ps_raw_subset, Group=="Placebo")
ps_raw_week0 <- subset_samples(ps_raw_subset, (Group == "Placebo" |  Group == "FMT") & week == "Week_0") 
ps_raw_week7 <- subset_samples(ps_raw_subset, (Group == "Placebo" |  Group == "FMT") & week == "Week_7") 
```

```{r, warning = FALSE, message = FALSE, echo = FALSE}

# Get the top50 most abundant KOs out of the filtered phyloseq object
top50 <- names(sort(taxa_sums(ps_subset), decreasing=TRUE)[1:50])
top50 # list of most abundant KOs

# Create a phyloseq object only with the top50 KOs
ps_top50 = prune_taxa(top50, ps_subset)



# List of KOs involved in the terminal steps of the Butyrate production
butyrate_list <- c("K00634", "K00929", "K01034", "K01035", "K19709", "K18122", "K14534", "K18014", "K01615", "K17865")

# List of KOs involved in the final terminal steps of the Butyrate production
last_butyrate <- c("K01034", "K01035", "K19709", "K00929", "K18122")

# Create a phyloseq object only with the Butyrate KOs
ps_butyrate = prune_taxa(butyrate_list, ps_subset)
ps_butyrate_FMT = prune_taxa(butyrate_list, ps_FMT)
ps_butyrate_placebo = prune_taxa(butyrate_list, ps_placebo)

ps_last_butyrate = prune_taxa(last_butyrate, ps_subset)
ps_last_butyrate_FMT = prune_taxa(last_butyrate, ps_FMT)
ps_last_butyrate_placebo = prune_taxa(last_butyrate, ps_placebo)

phyloseq::taxa_names(ps_butyrate) <- phyloseq::tax_table(ps_butyrate)[, "Symbol"]
phyloseq::taxa_names(ps_last_butyrate) <- phyloseq::tax_table(ps_last_butyrate)[, "Symbol"]

```
# First glance at results
##  Most abudant KOs 
```{r}

plot <- plot_bar(ps_top50, fill = "KO") + 
  geom_bar(aes(color=KO, fill=KO), stat="identity", position="fill") + 
  labs(x="Samples", y="Abundance", title="The 50 most abundant KO") +
  facet_wrap(~week, scales="free_x") +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot 

ps_top50_relab = phyloseq::transform_sample_counts(ps_top50, function(x){(x / sum(x))})
plot <- plot_bar(ps_top50_relab, fill = "KO") + 
  geom_bar(aes(color=KO, fill=KO), stat="identity", position="fill") + 
  labs(x="Samples", y="Relative abundance between Top50", title="The 50 most abundant KO") + 
  facet_wrap(~week, scales="free_x") +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot

```
## Butyrate production related KOs

```{r}

plot <- plot_bar(ps_butyrate, fill = "Symbol") + 
  facet_wrap(~group_week, scales="free_x")  +
  geom_bar(aes(color=Symbol, fill=Symbol), stat="identity", position="fill") + 
  labs(x="Samples", y="Abundance", title="Butyrate Synthesis related KOs") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot


plot <- plot_bar(ps_last_butyrate, fill = "Symbol") + 
  facet_wrap(~group_week, scales="free_x")  +
  geom_bar(aes(color=Symbol, fill=Symbol), stat="identity", position="fill") + 
  labs(x="Samples", y="Abundance", title="Last Butyrate Synthesis related KOs") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot

ps_butyrate_relab = phyloseq::transform_sample_counts(ps_butyrate, function(x){(x / sum(x))})
plot <- plot_bar(ps_butyrate_relab, fill = "Symbol") + 
  facet_wrap(~group_week, scales="free_x")  +
  geom_bar(aes(color=Symbol, fill=Symbol), stat="identity", position="fill") + 
  labs(x="Samples", y="Relative abundance between butyrate genes", title="Butyrate Synthesis related KOs") +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        plot.title = element_text(size = 30))
plot


plot <- plot_bar(ps_butyrate_FMT, fill = "Symbol") + 
  facet_wrap(~week, scales="free_x")  +
  geom_bar(aes(color=Symbol, fill=Symbol), stat="identity", position="fill") + 
  labs(x="Samples", y="Abundance", title="Butyrate Synthesis related KOs in FMT Samples") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot


plot <- plot_bar(ps_last_butyrate_FMT, fill = "Symbol") + 
  facet_wrap(~week, scales="free_x")  +
  geom_bar(aes(color=Symbol, fill=Symbol), stat="identity", position="fill") + 
  labs(x="Samples", y="Abundance", title="Last Butyrate Synthesis related KOs in FMT Samples") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))
plot



```

# Alpha Diversity

Next, we are going to calculate the Alpha Diversity of our samples. **Definition of alpha diversiy and estimates we are using**

To do so, we first need to add the diversity estimates to our phyloseq object, and test if the data are normally distributed using Shapiro-Wilk Normality test. In this case, will focus on the Shannon, Chao1 and Observed diversity estimates.

You want the data to be roughly normal so that you can run ANOVA or t-tests. If it is not normally distributed, you will need to consider non-parametric tests such as Kruskal-Wallis.

Overall, for alpha-diversity:

ANOVA, t-test, or general linear models with the normal distribution are used when the data is roughly normal
Kruskal-Wallis, Wilcoxon rank sum test, or general linear models with another distribution are used when the data is not normal. 

### Test Normality of the Shannon diversity estimations:

First, we need to extract the estimates of the alpha diversity and then perform a Shapiro-Wilk Normality test to see if this data follows a normal distribution. The null hypothesis of this test is that “sample distribution is normal”. If the test is significant, the distribution is non-normal. 

```{r}

# Extract Shannon results
alpha_results <- estimate_richness(ps_subset, measures = c("Shannon", "Observed", "Chao1"))

# Shapiro-Wilk Normality test.
# The null hypothesis of this test is that “sample distribution is normal”. If the test is significant, the distribution is non-normal. 
shapiro.test(alpha_results$Shannon)
shapiro.test(alpha_results$Observed)
shapiro.test(alpha_results$Chao1)
```

From the output, the p-value > 0.05 implying that the distribution of the data are not significantly different from normal distribution. In other words, we can assume the normality for all three alpha diversity estimates

### Plot Shannon diversity and test differences between groups. 

We can plot the Shannon Diversity of each sample group and test its significance with t-tests. 

```{r, warning = FALSE, message = FALSE, echo = FALSE}


symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))


# Plot all functional alpha diversity estimates colored by week
plot_richness(ps_subset, x="Sample_id", color="week", measures=c("Shannon", "Observed", "Chao1")) + 
  labs(title="Alpha Diversity of each Sample", x="Sample IDs") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))


#a_my_comparisons <- list(c("FMT_Week_0", "Placebo_Week_0"), c("FMT_Week_7", "Placebo_Week_7")) 
plot_richness(ps_subset, x="group_week", color="group_week", nrow = 2, measures=c("Shannon", "Observed", "Chao1")) + 
  #facet_wrap(~week, scales="free_x") + 
  geom_boxplot(aes(fill=group_week),alpha=0.2) + 
  labs(title="Alpha Diversity") +
  #stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE) + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))

b_my_comparisons <- list(c("FMT", "Placebo"), c("Donor", "FMT"), c("Placebo", "Donor"))
# Boxplot Shannon functional diversity grouped by Group and separated by Week
plot_richness(ps_subset, x="Group", color="Group", measures=c("Shannon")) + 
  facet_wrap(~week, scales="free_x") + 
  geom_boxplot(aes(fill=Group),alpha=0.2) + 
  labs(title="Shannon Index grouped by Subjects'") +
  stat_compare_means(method = "t.test", comparisons = b_my_comparisons, label = "p.signif", symnum.args = symnum.args) + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))

# Boxplot Observed functional diversity grouped by Group and separated by Week
plot_richness(ps_subset, x="Group", color="Group", measures=c("Observed")) + 
  facet_wrap(~week, scales="free_x") + 
  geom_boxplot(aes(fill=Group),alpha=0.2) + 
  labs(title="Observed diversity grouped by Subjects'") +
  stat_compare_means(method = "t.test", comparisons = b_my_comparisons, label = "p.signif", symnum.args = symnum.args) + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))



c_my_comparisons <- list( c("Week_0", "Week_7")) 
## Plot all Shannon functional alpha diversity grouped by week
plot_richness(ps_subset, color = "week", x = "week", measures = c("Shannon")) + 
  geom_boxplot(aes(fill = week), alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = c_my_comparisons, label = "p.signif", symnum.args = symnum.args) +
  labs(title="Shannon Index grouped by Week") + 
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))


## Plot FMT functional alpha diversity grouped by week

alpha_results_FMT <- estimate_richness(ps_FMT, measures = c("Shannon", "Observed"))

plot_richness(ps_FMT, color = "week", x = "week", measures = c("Shannon")) + 
  geom_boxplot(aes(fill = week), alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = c_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE) +
  labs(title="Shannon Index of FMT subjects grouped by Week") + 
  geom_line(aes(x = week, y = alpha_results_FMT$Shannon, group = id), color='black', alpha=0.1) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))


plot_richness(ps_FMT, color = "week", x = "week", measures = c("Observed")) + 
  geom_boxplot(aes(fill = week), alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = c_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE) +
  labs(title="Observed Diversity of FMT subjects grouped by Week") + 
  geom_line(aes(x = week, y = alpha_results_FMT$Observed, group = id), color='black', alpha=0.1) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))

## Plot Placebo functional alpha diversity grouped by week

alpha_results_placebo <- estimate_richness(ps_placebo, measures = c("Shannon", "Observed"))

plot_richness(ps_placebo, color = "week", x = "week", measures = c("Shannon")) + 
  geom_boxplot(aes(fill = week), alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = c_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE) +
  labs(title="Shannon Index of Placebo subjects grouped by Week") + 
  geom_line(aes(x = week, y = alpha_results_placebo$Shannon, group = id), color='black', alpha=0.1) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))

plot_richness(ps_placebo, color = "week", x = "week", measures = c("Observed")) + 
  geom_boxplot(aes(fill = week), alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = c_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE) +
  labs(title="Shannon Index of Placebo subjects grouped by Week") + 
  geom_line(aes(x = week, y = alpha_results_placebo$Observed, group = id), color='black', alpha=0.1) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 20),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        strip.text = element_text(size = 20),
        plot.title = element_text(size = 30))

```
# Beta Diversity

## Data preprocessing with DESeq2

We are going to normalize the raw phyloseq object using DESeq2. 

```{r Loading the data}
set.seed(0503)

# First we need to convert the raw phyloseq object to a deseq object
ps_dds <- phyloseq_to_deseq2(ps_raw_subset, ~Group + week )
# and running DEEeq2 standard analysis:
ps_dds <- DESeq(ps_dds)

# geometric mean, set to zero when all coordinates are zero
geo_mean_protected <- function(x) {
  if (all(x == 0)) {
    return (0)
  }
  exp(mean(log(x[x != 0])))
}

geoMeans <- apply(counts(ps_dds), 1, geo_mean_protected)
ps_dds <- estimateSizeFactors(ps_dds, geoMeans = geoMeans)
ps_dds <- estimateDispersions(ps_dds)
abund <- getVarianceStabilizedData(ps_dds)

# making our phyloseq object with transformed table
vst_count_phy <- otu_table(abund, taxa_are_rows=T)
sample_info_tab_phy <- sample_data(samples_df)
vst_physeq <- phyloseq(vst_count_phy, sample_info_tab_phy)

```

## Principal coordinate analysis (PCoA) of Bray-Curtis distances

### DESeq2 Transformed Data
```{r, warning = FALSE, message = FALSE, echo = FALSE, include=TRUE}

set.seed(0503)

# generating and visualizing the PCoA with phyloseq
vst_pcoa <- ordinate(vst_physeq, method="PCoA", distance="bray")
eigen_vals <- vst_pcoa$values$Eigenvalues # allows us to scale the axes according to their magnitude of separating apart the samples

plot_ordination(vst_physeq, vst_pcoa, color="Group") + 
  theme(aspect.ratio=1) + 
  labs(title = "PCoA (Bray-Curtis index)", subtitle = "DESeq2 Transformation") +
  stat_ellipse(aes(group = Group), linetype = 2) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        plot.title = element_text(size = 25),
        plot.subtitle = element_text(size = 20))

plot_ordination(vst_physeq, vst_pcoa, color="group_week") + 
  theme(aspect.ratio=1) + 
  labs(title = "PCoA (Bray-Curtis index)", subtitle = "DESeq2 Transformation") +
  stat_ellipse(aes(group = group_week), linetype = 2) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        plot.title = element_text(size = 25), 
        plot.subtitle = element_text(size = 20))

```
### Median Normalized data
```{r, warning = FALSE, message = FALSE, echo = FALSE, include=TRUE}
set.seed(0503)

# generating and visualizing the PCoA with phyloseq
vst_pcoa <- ordinate(ps_subset, method="PCoA", distance="bray")
eigen_vals <- vst_pcoa$values$Eigenvalues # allows us to scale the axes according to their magnitude of separating apart the samples

plot_ordination(ps_subset, vst_pcoa, color="Group") + 
  theme(aspect.ratio=1) + 
  labs(title = "PCoA (Bray-Curtis index)", subtitle = "Median Normalization") +
  stat_ellipse(aes(group = Group), linetype = 2) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        plot.subtitle = element_text(size = 20),
        plot.title = element_text(size = 25))


plot_ordination(ps_subset, vst_pcoa, color="group_week") + 
  theme(aspect.ratio=1) + 
  labs(title = "PCoA (Bray-Curtis index)", subtitle = "Median Normalization") +
  stat_ellipse(aes(group = group_week), linetype = 2) +
  theme(axis.text.x = element_text(size = 15),
        axis.text.y = element_text(size = 15),
        axis.title.x = element_text(size = 20),
        axis.title.y = element_text(size = 20),
        legend.title = element_text(size = 20), 
        legend.text = element_text(size = 15),
        plot.subtitle = element_text(size = 20),
        plot.title = element_text(size = 25))

```


## Betadisper and permutational ANOVA

Betadisper and permutational ANOVA 

Here we are going to test if there is a statistically signficant difference between our sample types. One way to do this is with the betadisper and adonis functions from the vegan package. adonis can tell us if there is a statistical difference between groups. However, adonis assumes that there is no statistical difference within the same group. To check whether this assumption is met, we first need to perfom a betadisper permutest. If the p-value is non-significant (p-value > 0.05), there is a sufficient level of homogeneity of dispersion within groups. If there is not, then the adonis permanova test can be unreliable.

### DESeq2 Transformed data

#### Group


```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Betadisper Permutest 
bray_dist = phyloseq::distance(vst_physeq, method="bray", weighted=T)

disp.age = betadisper(bray_dist, sample_data(vst_physeq)$Group)

disp.age$call

permutest(disp.age, pairwise=TRUE, permutations=1000)

```

The p-value of the betadisper permutest is higher than 0.05 so we can assume that sample groups are sufficiently homogenous and we can carry on with the adonis permanova test.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Adonis PERMANOVA analysis
adonis2(bray_dist ~ sample_data(vst_physeq)$Group)
```
The p-value of the Adonis permanova test is lower than 0.05 so there is a statistically significant difference between groups. 


#### group_week

```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Betadisper Permutest

bray_dist = phyloseq::distance(vst_physeq, method="bray", weighted=T)

disp.age = betadisper(bray_dist, sample_data(vst_physeq)$group_week)

disp.age$call

permutest(disp.age, pairwise=TRUE, permutations=1000)

```
The p-value of the betadisper permutest is higher than 0.05 so we can assume that sample groups are sufficiently homogenous and we can carry on with the adonis permanova test.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
# Adonis PERMANOVA analysis
adonis2(bray_dist ~ sample_data(vst_physeq)$group_week)
```
The p-value of the Adonis permanova test is higher than 0.05 so we can't prove that there is a statistically significant difference between groups. 

### Median Normalized data
#### Group


```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Betadisper Permutest 
bray_dist = phyloseq::distance(ps_subset, method="bray", weighted=T)

disp.age = betadisper(bray_dist, sample_data(ps_subset)$Group)

disp.age$call

permutest(disp.age, pairwise=TRUE, permutations=1000)

```

The p-value of the betadisper permutest is higher than 0.05 so we can assume that sample groups are sufficiently homogenous and we can carry on with the adonis permanova test.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Adonis PERMANOVA analysis
adonis2(bray_dist ~ sample_data(ps_subset)$Group)
```
The p-value of the Adonis permanova test is lower than 0.05 so there is a statistically significant difference between groups. 


#### group_week

```{r, warning = FALSE, message = FALSE, echo = FALSE}
set.seed(0503)

# Betadisper Permutest

bray_dist = phyloseq::distance(ps_subset, method="bray", weighted=T)

disp.age = betadisper(bray_dist, sample_data(ps_subset)$group_week)

disp.age$call

permutest(disp.age, pairwise=TRUE, permutations=1000)

```
The p-value of the betadisper permutest is higher than 0.05 so we can assume that sample groups are sufficiently homogenous and we can carry on with the adonis permanova test.

```{r, warning = FALSE, message = FALSE, echo = FALSE}
# Adonis PERMANOVA analysis
adonis2(bray_dist ~ sample_data(ps_subset)$group_week)
```
The p-value of the Adonis permanova test is higher than 0.05 so we can't prove that there is a statistically significant difference between groups. 



# Differential Abundance Analysis

## DESeq2 Differential Analysis with Microbial

Find features that are up or down represented in the compared groups using the microbial package that implements DESeq2. 
- The difftest function from the microbial package finds features that are up or down represented in the compared groups. The difftest function requires a phyloseq object containing merged information of abundance, taxonomic assignment and sample data including the measured variables and categorical information of the samples. Since the DESeq2 transformation is performed in this function, *RAW COUNT* values are preferred. In this context, we will be using the ps_raw phyloseq object and its subsets.  
- The plotdiff function produces a figure with the top most annotated features with corresponding adjusted p-values and abundance distribution with respect to control conditions. 


The group parameter is a character string specifying the name of a categorical variable containing grouping information. The pvalue and log2FC are thresholds for p values and log2 fold change. Adjusted p value cutoff is also supported by specify the padj paramater.

### According to Week Condition between the FMT subjects (Week 7 vs Week 0)

```{r, warning = FALSE, message = FALSE, echo = FALSE}

# alteradovscontrol

res <- difftest(ps_raw_FMT, group= "week", ref = "Week_0")
res$Phylum <- res$KO 

write.table(res,"results_deseq2_Week7vsWeek0_FMT_des.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Description", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for Week_7 vs Week_0 in FMT subjects \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


write.table(res,"results_deseq2_Week7vsWeek0_FMT_sym.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Symbol", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for Week_7 vs Week_0 in FMT subjects \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


```
### According to Week Condition between the Placebo subjects (Week 7 vs Week 0)

```{r, warning = FALSE, message = FALSE, echo = FALSE}

# alteradovscontrol

res <- difftest(ps_raw_placebo, group= "week", ref = "Week_0")
res$Phylum <- res$KO 

write.table(res,"results_deseq2_Week7vsWeek0_placebo_des.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Description", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for Week_7 vs Week_0 in Placebo subjects \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


write.table(res,"results_deseq2_Week7vsWeek0_placebo_sym.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Symbol", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for Week_7 vs Week_0 in Placebo subjects \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


```



### According to Group Condition between samples collected at Week 0 (FMT vs Placebo)

```{r, warning = FALSE, message = FALSE, echo = FALSE}

# alteradovscontrol

res <- difftest(ps_raw_week0, group= "Group", ref = "Placebo")
res$Phylum <- res$KO 

write.table(res,"results_deseq2_FMTvsPlacebo_week0_des.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Description", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for FMT vs Placebo subjects in samples collected at Week 0 \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


write.table(res,"results_deseq2_FMTvsPlacebo_week0_sym.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Symbol", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for FMT vs Placebo subjects in samples collected at Week 0 \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


```


### According to Group Condition between samples collected at Week 7 (FMT vs Placebo)

```{r, warning = FALSE, message = FALSE, echo = FALSE}

# alteradovscontrol

res <- difftest(ps_raw_week7, group= "Group", ref = "Placebo")
res$Phylum <- res$KO 

write.table(res,"results_deseq2_FMTvsPlacebo_week7_des.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Description", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for FMT vs Placebo subjects in samples collected at Week 7 \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


write.table(res,"results_deseq2_FMTvsPlacebo_week7_sym.tsv", sep = "\t", quote = F) 
plotdiff(res, level="Symbol", pvalue=0.05, log2FC = 2) + geom_hline(yintercept = 0, linetype="dotted")+ geom_point(size = 4) +
  labs(y = "\nLog2 Fold-Change for FMT vs Placebo subjects in samples collected at Week 7 \n pvalue = 0.05 , log2FC >= 2", x = "") +
  theme(axis.text.x = element_text(color = "black", size = 12),
        axis.text.y = element_text(color = "black", size = 12),
        axis.title.y = element_text(size = 14),
        axis.title.x = element_text(size = 14),
        legend.text = element_text(size = 12),
        legend.title = element_text(size = 12)) +
  geom_hline(yintercept = 0, linetype="dotted")


```


## Plots of Butyrate Production related genes with statistics tests
```{r}

symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))

a_my_comparisons <- list( c("Week_0", "Week_7"))

# Butyrate Production comparision between week 0 and week 7 in Placebo patients
phyloseq::taxa_names(ps_butyrate_placebo)<- phyloseq::tax_table(ps_butyrate_placebo)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_placebo) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Placebo Butyrate production") +
facet_wrap(~ OTU, scales = "free") + 
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)


phyloseq::taxa_names(ps_butyrate_FMT)<- phyloseq::tax_table(ps_butyrate_FMT)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_FMT) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "FMT Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)

phyloseq::taxa_names(ps_butyrate_FMT)<- phyloseq::tax_table(ps_butyrate_FMT)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_FMT) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "FMT Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)

b_my_comparisons <- list( c("Placebo", "FMT"))
b2_my_comparisons <- list( c("FMT", "Placebo"), c("Donor", "Placebo"), c("FMT", "Donor"))

ps_butyrate_week_0 <- subset_samples(ps_butyrate, week == "Week_0")
phyloseq::taxa_names(ps_butyrate_week_0)<- phyloseq::tax_table(ps_butyrate_week_0)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_week_0) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Week 0 Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b2_my_comparisons, label = "p.signif", symnum.args = symnum.args)

ps_butyrate_week_7 <- subset_samples(ps_butyrate, week == "Week_7")
phyloseq::taxa_names(ps_butyrate_week_7)<- phyloseq::tax_table(ps_butyrate_week_0)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_week_7) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Week 7 Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b_my_comparisons, label = "p.signif", symnum.args = symnum.args)


phyloseq::taxa_names(ps_butyrate)<- phyloseq::tax_table(ps_butyrate)[, "Symbol"]
phyloseq::psmelt(ps_butyrate) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Placebo Butyrate production") +
facet_wrap(~ OTU, scales = "free") + 
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b2_my_comparisons, label = "p.signif", symnum.args = symnum.args)


```

### Removing samples from one of the donors
```{r}

symnum.args = list(cutpoints = c(0, 0.0001, 0.001, 0.01, 0.05, 1), symbols = c("****", "***", "**", "*", "ns"))

a_my_comparisons <- list( c("Week_0", "Week_7"))

# Butyrate Production comparision between week 0 and week 7 in Placebo patients
phyloseq::taxa_names(ps_butyrate_placebo)<- phyloseq::tax_table(ps_butyrate_placebo)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_placebo) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Placebo Butyrate production") +
facet_wrap(~ OTU, scales = "free") + 
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)

ps_butyrate_FMT <- subset_samples(ps_butyrate_FMT, Donor != "Donor_485")
phyloseq::taxa_names(ps_butyrate_FMT)<- phyloseq::tax_table(ps_butyrate_FMT)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_FMT) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "FMT Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)

phyloseq::taxa_names(ps_butyrate_FMT)<- phyloseq::tax_table(ps_butyrate_FMT)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_FMT) %>%
ggplot(data = ., aes(x = week, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_point(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "FMT Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  geom_line(aes(x = week, y = Abundance, group = id),linetype="11", alpha=0.2) +
  stat_compare_means(method = "t.test", comparisons = a_my_comparisons, label = "p.signif", symnum.args = symnum.args, paired=TRUE)

b_my_comparisons <- list( c("Placebo", "FMT"))
b2_my_comparisons <- list( c("FMT", "Placebo"), c("Donor", "Placebo"), c("FMT", "Donor"))

ps_butyrate_week_0 <- subset_samples(ps_butyrate, week == "Week_0" & Donor != "Donor_485")
phyloseq::taxa_names(ps_butyrate_week_0)<- phyloseq::tax_table(ps_butyrate_week_0)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_week_0) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Week 0 Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b2_my_comparisons, label = "p.signif", symnum.args = symnum.args)

ps_butyrate_week_7 <- subset_samples(ps_butyrate, week == "Week_7")
phyloseq::taxa_names(ps_butyrate_week_7)<- phyloseq::tax_table(ps_butyrate_week_0)[, "Symbol"]
phyloseq::psmelt(ps_butyrate_week_7) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +
geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = OTU), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Week 7 Butyrate Production") +
facet_wrap(~ OTU, scales = "free") +
  theme(legend.position="none", axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b_my_comparisons, label = "p.signif", symnum.args = symnum.args)


phyloseq::taxa_names(ps_butyrate)<- phyloseq::tax_table(ps_butyrate)[, "Symbol"]
phyloseq::psmelt(ps_butyrate) %>%
ggplot(data = ., aes(x = Group, y = Abundance)) +geom_boxplot(outlier.shape = NA) +
geom_jitter(aes(color = Donor), height = 0, width = .2) +
labs(x = "", y = "Abundance\n", title = "Placebo Butyrate production") +
facet_wrap(~ OTU, scales = "free") + 
  theme(axis.text.x=element_text(angle=45,hjust=1,vjust=1,size=12))+
  stat_compare_means(method = "t.test", comparisons = b2_my_comparisons, label = "p.signif", symnum.args = symnum.args)


```



